### DISCUSSION TITLE
PDCursesMod 4.3.7

### DISCUSSION THREAD

**Discussion by Bill-Gray** [Bill-Gray/PDCursesMod#297] [Announcements]
PDCursesMod 4.3.7 - 2023 June 17
================================

(See [`docs/HISTORY.md `](https://github.com/Bill-Gray/PDCursesMod/blob/master/docs/HISTORY.md) for more details, and/or the [Git log](https://github.com/Bill-Gray/PDCursesMod/commits/v4.3.7) for extreme detail.)

Major new features
------------------

- SDL2 GL port,  from Julius Ikkala.  This is (on most systems) faster than the "traditional" SDL2 port. Many commits,  starting at 4678131a39.

Minor new features
------------------

- Support on Windows for Unicode input in the SMP (codepoints above 0xFFFF, for instance emojis.) 2050706649  75b62ea015

- WinCon: supports full unicode output including SMP.  3ec63c9067 7089b2eae1  fac87bd938

- WinCon, WinGUI can be cross-compiled for Windows on Arm.  158e8ae5b7

- 'newtest': extended the SLK testing.  'newtest' and 'testcurs' now use wget_wch( ) in wide builds,  so the actual character is shown on input tests (instead of the multibyte stream).  'testcurs' gradient example (showing colors beyond 256 palette entries) is a little more logically set out.  c744e3ec03 7089b57d07

- WinCon can support a 'default' (transparent) background.  81cbefbd4b

- 'curspriv.h' modified to allow C++ compilation.  52a2fdf518

- Clarified compatibility for ACS_ and WACS_ box characters,  and added  many missing aliases.  From a suggestion by Simon Sobisch.  e8f7b33bbf

- Fixed warnings with OpenWATCOM.  We now compile with -we ("treat warnings as errors").  f27369e93b  abbd8f5268  89e3426248

- If you try to compile the DOS version with WIDE=Y,  you get an error message explaining that wide characters don't work on that platform.   d07b884502

- Now using ncurses-6.4 tests instead of 6.3.

Bug fixes
---------

- WinCon: Key modifiers were incorrectly cleared without a key event.  9229f08604

- VT: Ctrl-Alt-letter input was off by one.  Hit,  say,  Ctrl-Alt-T, and you'd get Ctrl-Alt-U.  65568582af

- Fixed some casting issues with GetProcAddress( ) in Windows and a deprecation warning with MinGW,  with considerable help from Arnold Trembley and Chuck Haatvedt.  Further work may be required (it works,  but pedantic compilation can get you warnings).  bbae4bb0e8  28cbcfcc34

- CMake's list of demos is now up to date for DOS,  DOSVGA,  WinCon,  WinGUI.  Others to follow.  4f3dc96436

- Pasting text from the clipboard was done with a buffer lacking room for a null terminator,  resulting in a memory overwrite.  Reported by nhmall (issue #295).  0a4e1e9065

- realloc( ptr, 0); was assumed to free ptr and return NULL.  This is usually true,  but isn't guaranteed and isn't true on FreeBSD,  and caused an assert to be triggered in endwin().  abbb6de7de

- 'terminfo' functions were not included in the DOSVGA builds,  nor in WinCon when using the Microsoft VC makefile.  Also,  the WinCon Digital Mars makefile was broken.  c51efbbf88

- 'picsview' and 'mbrot' demos now work correctly with ncurses.  Some unnecessary distinctions between PDCurses* and ncurses were removed.  89727917e7

<hr /><em>This discussion was created from the release <a href='https://github.com/Bill-Gray/PDCursesMod/releases/tag/v4.3.7'>PDCursesMod 4.3.7</a>.</em>

**Comment by GitMensch** [comment]
PR to update it in MSYS2 packages (and to enable the vt port, which will be build and installed as an additional library with its port name as suffix for testing purposes): https://github.com/msys2/MINGW-packages/pull/17567 - now merged and also binaries available.
